adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -77,6 +86,10 @@ String getTenantId() {
      * @return The device identifier.
      */
     String getDeviceId() {
+        if (receiverTarget == null) {
```

Comment:
don't we have all necessary information available in the constructor to properly initialize the fields? Why do we need to do additional computations in the accessors? This class is supposed to be immutable, isn't it?
Doing all initializaiton and computation in the constructor would help very much with documenting and understanding the different cases that are possible, don't you think?